node/pkg/driver/device_connectivity/device_connectivity_fc.go

```diff
@@ -0,0 +1,324 @@
+
+//go:generate mockgen -destination=../../../mocks/mock_OsDeviceConnectivityHelperFcInterface.go -package=mocks github.com/ibm/ibm-block-csi-driver/node/pkg/driver/device_connectivity OsDeviceConnectivityHelperFcInterface
+
+type OsDeviceConnectivityHelperFcInterface interface {
+	/*
+		This is helper interface for OsDeviceConnectivityFc.
+		Mainly for writting clean unit testing, so we can Mock this interface in order to unit test OsDeviceConnectivityFc logic.
+	*/
+	WaitForPathToExist(devicePath string, maxRetries int, intervalSeconds int) ([]string, bool, error)
+	GetMultipathDisk(path string) (string, error)
+	GetFcHostIDs() ([]int, error)
+}
+
+type OsDeviceConnectivityHelperFc struct {
+	executer executer.ExecuterInterface
+}
+
+const (
+	FC_HOST_SYSFS_PATH = "/sys/class/fc_host"
+)
+
+func NewOsDeviceConnectivityHelperFc(executer executer.ExecuterInterface) OsDeviceConnectivityHelperFcInterface {
+	return &OsDeviceConnectivityHelperFc{executer: executer}
+}
+
+func (o OsDeviceConnectivityHelperFc) GetFcHostIDs() ([]int, error) {
+	/*
+		Description:
+			This function find all the hosts IDs under directory /sys/class/fc_host/"
+			So the function goes over all the above hosts and return back only the host numbers as a list.
+	*/
+
+	targetNamePath := FC_HOST_SYSFS_PATH + "/host*/port_state"
+	var HostIDs []int
+	matches, err := o.executer.FilepathGlob(targetNamePath)
+	if err != nil {
+		logger.Errorf("Error while Glob targetNamePath : {%v}. err : {%v}", targetNamePath, err)
+		return HostIDs, err
+	}
+
+	logger.Debugf("targetname files matches were found : {%v}", matches)
+
+	for _, targetPath := range matches {
+		logger.Debugf("Check if targetfile (%s) value is Online.", targetPath)
+		targetName, err := o.executer.IoutilReadFile(targetPath)
+		if err != nil {
+			logger.Warningf("Could not read target name from file : {%v}, error : {%v}", targetPath, err)
+			continue
+		}
+
+		if strings.EqualFold(strings.TrimSpace(string(targetName)), "online") {
+			re := regexp.MustCompile("host([0-9]+)")
+			regexMatch := re.FindStringSubmatch(targetPath)
+			logger.Tracef("Found regex matches : {%v}", regexMatch)
+			hostNumber := -1
+
+			if len(regexMatch) < 2 {
+				logger.Warningf("Could not find host number for targetNamePath : {%v}", targetPath)
+				continue
+			} else {
+				hostNumber, err = strconv.Atoi(regexMatch[1])
+				if err != nil {
+					logger.Warningf("Host number in for target file was not valid : {%v}", regexMatch[1])
+					continue
+				}
+			}
+
+			HostIDs = append(HostIDs, hostNumber)
+			logger.Debugf("targetname path (%s) was found. Adding host number {%v} to the session list.", targetPath, hostNumber)
+
+		}
+	}
+
+	if len(HostIDs) == 0 {
+		return []int{}, &ConnectivityFcHostTargetNotFoundError{DirectoryPath: FC_HOST_SYSFS_PATH}
+	}
+	return HostIDs, nil
+
+}
+
+func (o OsDeviceConnectivityHelperFc) WaitForPathToExist(devicePath string, maxRetries int, intervalSeconds int) ([]string, bool, error) {
```

Comment:
if this function is the same as iscsi helper 'OsDeviceConnectivityHelperIscsi) WaitForPathToExist'
then please refactor and move it to generic scsi interface with param of the globs to check.
again, reduce code and less UT

Reply:
Yes, I will refactor it

Reply:
if you can please do it as part of this PR.